### PR TITLE
fix: expose the volume spec topology through the openapi

### PR DESCRIPTION
**Pull request by tiagolobocastro**
Also allow to specify the volume labels on creation and make them a hashmap.

### CODE REVIEW

**Comment by tiagolobocastro** [comment]
bors try

File: common/src/types/v0/message_bus/volume.rs [hunk 1]
```diff
@@ -258,35 +270,38 @@ impl From<models::ExplicitTopology> for ExplicitTopology {
         }
     }
 }
+impl From<ExplicitTopology> for models::ExplicitTopology {
+    fn from(src: ExplicitTopology) -> Self {
+        Self::new(src.allowed_nodes, src.preferred_nodes)
+    }
+}
 
 /// Volume Healing policy used to determine if and how to replace a replica
```
Comment (paulyoong) [inline common/src/types/v0/message_bus/volume.rs:279]: Can remove "Healing".

File: control-plane/rest/src/versions/v0.rs [hunk 2]
```diff
@@ -148,19 +149,21 @@ pub struct CreateVolumeBody {
     pub size: u64,
     /// number of storage replicas
     pub replicas: u64,
-    // docs will be auto generated from the actual types
-    #[allow(missing_docs)]
-    pub policy: VolumeHealPolicy,
-    #[allow(missing_docs)]
-    pub topology: Topology,
+    /// Volume Healing policy used to determine if and how to replace a replica
```
Comment (paulyoong) [inline control-plane/rest/src/versions/v0.rs:152]: Remove "Healing".

**Review by paulyoong — APPROVED** [review]

File: control-plane/rest/src/versions/v0.rs (hunk 2 above)
Comment (tiagolobocastro) [reply control-plane/rest/src/versions/v0.rs]: Done for both :+1:

**Comment by tiagolobocastro** [comment]
bors merge

**Pull request merged**